Implementation smell,Namespace,Class,File,Method,Description
Long Parameter List,Owin,StartupExtensions,C:\research\architectureSmells\repos\owin_museum-piece-gate\src\Gate.Extensions\App_Packages\Owin.Extensions.Sources.0.6.4\StartupExtensions.cs,UseFunc,The method has 5 parameters. Parameters: builder' middleware' arg1' arg2' arg3
Long Parameter List,Owin,StartupExtensions,C:\research\architectureSmells\repos\owin_museum-piece-gate\src\Gate.Extensions\App_Packages\Owin.Extensions.Sources.0.6.4\StartupExtensions.cs,UseFunc,The method has 6 parameters. Parameters: builder' middleware' arg1' arg2' arg3' arg4
Long Parameter List,Owin,StartupExtensions,C:\research\architectureSmells\repos\owin_museum-piece-gate\src\Gate.Extensions\App_Packages\Owin.Extensions.Sources.0.6.4\StartupExtensions.cs,UseFunc,The method has 5 parameters. Parameters: builder' middleware' arg1' arg2' arg3
Long Parameter List,Owin,StartupExtensions,C:\research\architectureSmells\repos\owin_museum-piece-gate\src\Gate.Extensions\App_Packages\Owin.Extensions.Sources.0.6.4\StartupExtensions.cs,UseFunc,The method has 6 parameters. Parameters: builder' middleware' arg1' arg2' arg3' arg4
Long Identifier,System.Threading.Tasks,TaskHelpersExtensions,C:\research\architectureSmells\repos\owin_museum-piece-gate\src\Gate.Extensions\App_Packages\TaskHelpers.Sources.0.2\TaskHelpersExtensions.cs,,The length of the parameter _rethrowWithNoStackLossDelegate is 31.
Long Statement,System.Threading.Tasks,TaskHelpersExtensions,C:\research\architectureSmells\repos\owin_museum-piece-gate\src\Gate.Extensions\App_Packages\TaskHelpers.Sources.0.2\TaskHelpersExtensions.cs,CatchImpl,The length of the statement  "                            throw new InvalidOperationException("You must set the Task property of the CatchInfo returned from the TaskHelpersExtensions.Catch continuation.");" is 147.
Long Statement,System.Threading.Tasks,TaskHelpersExtensions,C:\research\architectureSmells\repos\owin_museum-piece-gate\src\Gate.Extensions\App_Packages\TaskHelpers.Sources.0.2\TaskHelpersExtensions.cs,CatchImplContinuation,The length of the statement  "            task.ContinueWith(innerTask => tcs.TrySetFromTask(innerTask)' TaskContinuationOptions.NotOnFaulted | TaskContinuationOptions.ExecuteSynchronously);" is 147.
Long Statement,System.Threading.Tasks,TaskHelpersExtensions,C:\research\architectureSmells\repos\owin_museum-piece-gate\src\Gate.Extensions\App_Packages\TaskHelpers.Sources.0.2\TaskHelpersExtensions.cs,CatchImplContinuation,The length of the statement  "                                throw new InvalidOperationException("You cannot return null from the TaskHelpersExtensions.Catch continuation. You must return a valid task or throw an exception.");" is 165.
Long Statement,System.Threading.Tasks,TaskHelpersExtensions,C:\research\architectureSmells\repos\owin_museum-piece-gate\src\Gate.Extensions\App_Packages\TaskHelpers.Sources.0.2\TaskHelpersExtensions.cs,CatchImplContinuation,The length of the statement  "                            throw new InvalidOperationException("You cannot return null from the TaskHelpersExtensions.Catch continuation. You must return a valid task or throw an exception.");" is 165.
Long Statement,System.Threading.Tasks,TaskHelpersExtensions,C:\research\architectureSmells\repos\owin_museum-piece-gate\src\Gate.Extensions\App_Packages\TaskHelpers.Sources.0.2\TaskHelpersExtensions.cs,GetRethrowWithNoStackLossDelegate,The length of the statement  "                // No-ops if the task completed successfully' else throws the originating exception complete with the correct call stack." is 121.
Empty Catch Block,System.Threading.Tasks,TaskHelpersExtensions,C:\research\architectureSmells\repos\owin_museum-piece-gate\src\Gate.Extensions\App_Packages\TaskHelpers.Sources.0.2\TaskHelpersExtensions.cs,GetRethrowWithNoStackLossDelegate,The method has an empty catch block.
Missing Default,System.Threading.Tasks,TaskHelpers,C:\research\architectureSmells\repos\owin_museum-piece-gate\src\Gate.Extensions\App_Packages\TaskHelpers.Sources.0.2\TaskHelpers.cs,SetIfTaskFailed,The following switch statement is missing a default case: switch (source.Status)             {                 case TaskStatus.Canceled:                 case TaskStatus.Faulted:                     return tcs.TrySetFromTask(source);             }
Missing Default,System.Threading.Tasks,TaskHelpersExtensions,C:\research\architectureSmells\repos\owin_museum-piece-gate\src\Gate.Extensions\App_Packages\TaskHelpers.Sources.0.2\TaskHelpersExtensions.cs,CopyResultToCompletionSourceImpl,The following switch statement is missing a default case: switch (task.Status)                 {                     case TaskStatus.Canceled:                     case TaskStatus.Faulted:                         TaskHelpers.TrySetFromTask(tcs' task);                         break;                      case TaskStatus.RanToCompletion:                         tcs.TrySetResult(resultThunk(task));                         break;                 }
Missing Default,System.Threading.Tasks,TaskHelpersExtensions,C:\research\architectureSmells\repos\owin_museum-piece-gate\src\Gate.Extensions\App_Packages\TaskHelpers.Sources.0.2\TaskHelpersExtensions.cs,CopyResultToCompletionSourceImplContinuation,The following switch statement is missing a default case: switch (innerTask.Status)                 {                     case TaskStatus.Canceled:                     case TaskStatus.Faulted:                         TaskHelpers.TrySetFromTask(tcs' innerTask);                         break;                      case TaskStatus.RanToCompletion:                         tcs.TrySetResult(resultThunk(task));                         break;                 }
